### PR TITLE
Feature/search resto

### PR DESCRIPTION
**Pull request by jasonevangelista**
- working search functionality (using landing page search bar and searchfilter page search bar)
- improved loading time from landing page search to searchfilter page w/ results (using next/router)
- search bar accepts any string except null or whitespace-only strings
- also added redirect to landing page when clicking zarap logo in header

### CODE REVIEW

File: pages/searchfilter/[searchitem].js [hunk 1]
```diff
@@ -1,109 +1,110 @@
...
+      />
+
+      <Row>
+        <Col span={6} className={styles.filterLayout}>
+          <FilterSection></FilterSection>
+        </Col>
+        <Col span={18} className={styles.cardsLayout}>
+          <h2>{resultStatus}</h2>
+          { cards }
+        </Col>
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js:63]: I suggest we leave out the filter component first and make the result cards bigger first as our goal is to make it ready for production. Try commenting out both columns and this column in the meantime.

`        <Col span={24} className={styles.cardsLayout}>
          <h2>{resultStatus}</h2>
          { cards }
        </Col>`

File: pages/searchfilter/[searchitem].js [hunk 2]
```diff
@@ -1,109 +1,110 @@
...
+import {Row, Col} from 'antd'
+import Search from 'antd/lib/input/Search'
+
+import { connectToDatabase } from '../../util/mongodb'
+import { useState } from 'react'
+import { useRouter } from 'next/router'
+
+export default function SearchFilter({ results }) {
+  const router = useRouter();
+  const { searchitem } = router.query
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Let's place semicolons statements like these

File: pages/searchfilter/[searchitem].js [hunk 3]
```diff
@@ -1,109 +1,110 @@
...
+
+  const [restaurants, setRestaurants] = useState(results);
+  const [resultStatus, setResultStatus] = useState(getDefaultResultStatus(restaurants));
+
+  const cards = generateRestaurantCards(restaurants)
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Same with concern above

File: pages/searchfilter/[searchitem].js [hunk 4]
```diff
@@ -1,109 +1,110 @@
...
+
+  return (
+
+    <div className={styles.pageLayout}>
+
+    <h1>Search Results on "{searchitem}"</h1>
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Indent here

File: pages/searchfilter/[searchitem].js [hunk 5]
```diff
@@ -1,109 +1,110 @@
...
-//   }
-// }
\ No newline at end of file
+    </div>
+  )
+}
+
+// generate each restaurant card based on results
+function generateRestaurantCards(restaurants){
+  var cards = []
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Semicolon

File: pages/searchfilter/[searchitem].js [hunk 6]
```diff
@@ -1,109 +1,110 @@
...
+  for (const [index, value] of restaurants.entries()){
+    cards.push(<SearchRestoCard key={index} resto={value}></SearchRestoCard>)
+  }
+  return cards
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Semicolon

File: pages/searchfilter/[searchitem].js [hunk 7]
```diff
@@ -1,109 +1,110 @@
...
\ No newline at end of file
+    </div>
+  )
+}
+
+// generate each restaurant card based on results
+function generateRestaurantCards(restaurants){
+  var cards = []
+  for (const [index, value] of restaurants.entries()){
+    cards.push(<SearchRestoCard key={index} resto={value}></SearchRestoCard>)
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Semicolon

File: pages/searchfilter/[searchitem].js [hunk 8]
```diff
@@ -1,109 +1,110 @@
...
+  }
+  return cards
+}
+
+// return default result status based on restaurant result length
+function getDefaultResultStatus(restaurants){
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Let's make it uniform to put spaces after parameter closing parenthesis  ) {

File: pages/searchfilter/[searchitem].js [hunk 9]
```diff
@@ -1,109 +1,110 @@
...
+  if(restaurants.length == 0){
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Same with the concern above

File: pages/searchfilter/[searchitem].js [hunk 10]
```diff
@@ -1,109 +1,110 @@
...
+   return 'No results found'
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Semicolon

File: pages/searchfilter/index.js [hunk 11]
```diff
@@ -6,23 +6,16 @@ import Search from 'antd/lib/input/Search'
...
+  const [restaurants, setRestaurants] = useState(results);
 
   // generate each restaurant card based on results
   const cards = []
-  // for (const [index, value] of restaurants.entries()){
-  //   cards.push(<SearchRestoCard key={index} resto={value}></SearchRestoCard>)
-  // }
-
-  for (const [index, value] of results.entries()){
+  for (const [index, value] of restaurants.entries()){
```
Comment (NeilMatthewLua) [inline pages/searchfilter/index.js]: Place whitespace before the curly braces

File: pages/searchfilter/[searchitem].js [hunk 12]
```diff
@@ -1,109 +1,110 @@
...
+
+  return {
+    props: {
+      results: JSON.parse(JSON.stringify(restaurants)),
+    },
+  };
+}
+
+// get restaurants based on search string
+async function getSearchResults(searchString){
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: Whitespace before curly braces

File: pages/searchfilter/[searchitem].js [hunk 13]
```diff
@@ -1,109 +1,110 @@
...
-
-//       <Row>
-//         <Col span={6} className={styles.filterLayout}>
-//           <FilterSection></FilterSection>
-//         </Col>
-
-//         <Col span={18} className={styles.cardsLayout}>
-//           <h2>{resultStatus}</h2>
-//           { cards }
+import styles from '../../styles/searchfilter/SearchFilter.module.css'
```
Comment (NeilMatthewLua) [inline pages/searchfilter/[searchitem].js]: It seems like for the other files, semicolons were used at the end of import statements. I leave it to you to decide if we should do that or not. Any is okay as long as it's uniform.

**Review by NeilMatthewLua — COMMENTED** [review]
Please see the following comments. Thanks!

**Review by Kurotwo — APPROVED** [review]

**Pull request merged**